File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int cs=nums[0];
        int maxsum=nums[0];
        for(int i=1;i<nums.length;i++){
            cs=Math.max(nums[i],cs+nums[i]);
            maxsum=Math.max(cs,maxsum);
        }
        return maxsum;
    }
}

File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int cs=0,ms=nums[0];
        for(int i=0;i<nums.size();i++){
            cs=max(cs+nums[i],nums[i]);
            ms=max(ms,cs);
        }
        return ms;
    }
};